game/world/objects/npc.cpp

```diff
@@ -2823,13 +2799,40 @@ void Npc::commitSpell() {
 const Npc::Routine& Npc::currentRoutine() const {
   auto time = owner.time();
   time = gtime(int32_t(time.hour()),int32_t(time.minute()));
-  for(auto& i:routines){
+  for(auto& i:routines) {
+    if(i.point==nullptr)
+      continue;
     if(i.end<i.start && (time<i.end || i.start<=time))
       return i;
     if(i.start<=time && time<i.end)
       return i;
     }
 
+  static std::set<int32_t> s;
+  if(routines.size()>0 && s.find(hnpc->id)==s.end()) {
+    s.insert(hnpc->id);
+    auto sym = owner.script().findSymbol(*hnpc);
+    Log::e("no valid routine found [",sym->name(),", time: ",time.hour(),":",time.minute(),"]");
```

Comment:
Any practical reason of `Log::e` here? Invalid routines are happening in normal play, and logging it as error will confuse someone who reads the log.
Probably can skip logging at all

Reply:
Removed the logging part.
>Any practical reason of Log::e here?

 Npc becomes stuck if invalid routine is queried and no fallback is present. I'd call accessing invalid routine an error. If error is the wrong word could call it warning like phoenix has. Was merely meant to see where buggy routines are because you don't notice them ingame now.